docs/Advanced-Usage/Chip-Communication.rst

```diff
@@ -0,0 +1,147 @@
+.. _chip-communication:
+
+Communicating with the Chip/DUT
+===============================
+
+There are two types of designs that can be
+made: tethered or standalone designs. A tethered design is where a host must interact with the DUT (a target) to bringup the design.
+A standalone design is a design that can bringup itself (has its own bootrom, loads programs itself, etc). An example of a tethered design
+is a Chipyard simulation where the host computer loads the test program into the designs memory. An example of a standalone design is
+a design where a program can be loaded from an SDCard by default.
+
+There are two ways the outside world (or host) can communicate with a tethered Chipyard design:
+
+* using the Tethered Serial Interface (TSI) or the Debug Module Interface (DMI) with the Front-End Server (FESVR) to communicate with the design
+* using the JTAG interface with OpenOCD and GDB to communicate with the design
+
+Using the Tethered Serial Interface (TSI) or the Debug Module Interface (DMI)
+-----------------------------------------------------------------------------
+
+If you are using TSI or the DMI to communicate with the target (DUT/chip), you are using
+the Front-End Server (FESVR) to facilitate communication between the host machine and the target.
+
+Primer on the Front-End Server (FESVR)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+FESVR is a C++ library that manages communication
+between a host machine and a RISC-V target. For debugging, it provides a simple API to reset,
+send messages, and load/run programs on a DUT. It also emulates peripheral devices.
+It can be incorporated with simulators (VCS, Verilator, FireSim), or used in a bringup sequence
+for a taped out chip.
+
+Specifically, FESVR uses the Host Target Interface (HTIF), a communication bus for the hardware,
+to speak with the target. HTIF is a non-standard Berkeley extension that uses a FIFO non-blocking
+interface to communicate with the target.
+
+Using the Tethered Serial Interface (TSI)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, Chipyard uses the Tethered Serial Interface (TSI) to communicate with the target.
```

Comment:
Should probably be consistent with the naming. Design (used further up in the document) vs. target. Keep in mind that firesim/midas use the term "target" to indicate the entire simulation (including models)